Bottom: c26eff41fa1f294646f7b471bfb8a8363decb270
Top:    610cbd7aaf6030c4a21c55de970200521aa8fc9b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-18 17:17:20 +0000

polygon1.c: Temporarily disable the seg_in_region test

The code does not handle the bounding region of arcs yet


---

diff --git a/src/polygon1.c b/src/polygon1.c
index bfb2637..e3b64b4 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -673,6 +673,7 @@ adjust_tree (rtree_t * tree, struct seg *s)
   return 0;
 }
 
+#if 0  /* TURNS OUT DISABLING THIS ACTUALLY (VERY MARGINALLY) SPEEDS THINGS UP! */
 /*
  * seg_in_region()
  * (C) 2006, harry eaton
@@ -694,6 +695,7 @@ seg_in_region (const BoxType * b, void *cl)
     return 0;
   return 1;			/* might intersect */
 }
+#endif
 
 /* Prepend a deferred node-insersion task to a list */
 static insert_node_task *
@@ -917,7 +919,7 @@ contour_bounds_touch (const BoxType * b, void *cl)
       info.tree = rtree_over->tree;
       if (info.tree)
 	if (UNLIKELY (r_search (info.tree, &info.s->box,
-				seg_in_region, seg_in_seg, &info)))
+				NULL /* seg_in_region */, seg_in_seg, &info)))
 	  assert (0); /* XXX: Memory allocation failure */
     }
   while ((av = NEXT_EDGE (av)) != &looping_over->head);
